core/src/providers/google_vertex_ai.rs

```diff
@@ -0,0 +1,527 @@
+    content: Content,
+    finish_reason: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct Completion {
+    candidates: Vec<Candidate>,
+    usage_metadata: Option<UsageMetadata>,
+}
+
+pub struct GoogleVertexAiProvider {}
+
+impl GoogleVertexAiProvider {
+    pub fn new() -> Self {
+        GoogleVertexAiProvider {}
+    }
+}
+
+#[async_trait]
+impl Provider for GoogleVertexAiProvider {
+    fn id(&self) -> ProviderID {
+        ProviderID::GoogleVertexAi
+    }
+
+    fn setup(&self) -> Result<()> {
+        utils::info("You cannot setup Google Vertex AI from the CLI, sorry.");
+
+        Ok(())
+    }
+
+    async fn test(&self) -> Result<()> {
+        Err(anyhow!(
+            "You cannot test Google Vertex AI from the CLI, sorry."
+        ))
+    }
+
+    fn llm(&self, id: String) -> Box<dyn LLM + Sync + Send> {
+        Box::new(GoogleVertexAiLLM::new(id))
+    }
+
+    fn embedder(&self, _id: String) -> Box<dyn Embedder + Sync + Send> {
+        unimplemented!()
+    }
+}
+
+pub struct GoogleVertexAiLLM {
+    id: String,
+    uri: Option<String>,
+    service_account_json: Option<String>,
+}
+
+impl GoogleVertexAiLLM {
+    pub fn new(id: String) -> Self {
+        GoogleVertexAiLLM {
+            id,
+            uri: None,
+            service_account_json: None,
+        }
+    }
+
+    fn tokenizer(&self) -> Arc<Mutex<CoreBPE>> {
+        cl100k_base_singleton()
+    }
+}
+
+#[async_trait]
+impl LLM for GoogleVertexAiLLM {
+    fn id(&self) -> String {
+        self.id.clone()
+    }
+
+    async fn initialize(&mut self, credentials: Credentials) -> Result<()> {
+        match credentials.get("GOOGLE_VERTEX_AI_SERVICE_ACCOUNT_JSON") {
+            Some(service_account_json) => {
+                self.service_account_json = Some(service_account_json.clone());
+            }
+            None => Err(anyhow!(
+                "GOOGLE_VERTEX_AI_SERVICE_ACCOUNT_JSON not found in credentials"
+            ))?,
+        }
+        match credentials.get("GOOGLE_VERTEX_AI_ENDPOINT") {
+            Some(uri) => {
+                self.uri = Some(uri.clone());
+            }
+            None => Err(anyhow!("GOOGLE_VERTEX_AI_URI not found in credentials"))?,
+        }
+        Ok(())
+    }
+
+    fn context_size(&self) -> usize {
+        8192
+    }
+
+    async fn encode(&self, text: &str) -> Result<Vec<usize>> {
+        encode_async(self.tokenizer(), text).await
+    }
+
+    async fn decode(&self, tokens: Vec<usize>) -> Result<String> {
+        decode_async(self.tokenizer(), tokens).await
+    }
+
+    async fn tokenize(&self, text: &str) -> Result<Vec<(usize, String)>> {
+        tokenize_async(self.tokenizer(), text).await
+    }
+
+    async fn generate(
+        &self,
+        prompt: &str,
+        mut max_tokens: Option<i32>,
+        temperature: f32,
+        n: usize,
+        stop: &Vec<String>,
+        _frequency_penalty: Option<f32>,
```

Comment:
Should the code throw an error for unsupported parameters to prevent misuse and enhance API clarity?